examples/pod-security-groups/index.ts

```diff
@@ -0,0 +1,196 @@
+configureClusterAccess("caller", cluster, callerSg);
+// Allow the caller job to access the nginx service
+new aws.vpc.SecurityGroupIngressRule("caller-to-nginx", {
+  securityGroupId: nginxSg.id,
+  ipProtocol: "tcp",
+  fromPort: 80,
+  toPort: 80,
+  referencedSecurityGroupId: callerSg.id,
+});
+
+// Assign the caller SecurityGroup to the caller job
+const callerSgp = new k8s.apiextensions.CustomResource("caller-sgp", {
+  apiVersion: "vpcresources.k8s.aws/v1beta1",
+  kind: "SecurityGroupPolicy",
+  metadata: {
+    name: "caller-sgp",
+  },
+  spec: {
+    podSelector: { matchLabels: { app: "caller" } },
+    securityGroups: {
+      groupIds: [callerSg.id],
+    }
+  },
+}, { provider: kube });
+
+// Create a job that is allowed to curl the nginx service. The job will fail if it can't reach the service.
+new k8s.batch.v1.Job("caller", {
+  spec: {
+    template: {
+      metadata: {
+        name: "caller",
+        labels: {
+          app: "caller",
+        },
+      },
+      spec: {
+        containers: [{
+          name: "caller",
+          image: "curlimages/curl",
+          command: ["curl", "--silent", "--show-error", "--fail", pulumi.interpolate`${nginx.metadata.name}.${nginx.metadata.namespace}:80`],
+        }],
+        restartPolicy: "Never",
+      },
+    },
+    backoffLimit: 3,
+  },
+}, { provider: kube, dependsOn: [nginx, callerSgp] });
+
+// Create a SecurityGroup for the wrongCaller job
+const wrongCallerSg = new aws.ec2.SecurityGroup("wrong-caller", {
+  vpcId: eksVpc.vpcId,
+});
+// Allow all traffic between the cluster and the caller SecurityGroup
+configureClusterAccess("wrong-caller", cluster, wrongCallerSg);
+
+// Assign the caller SecurityGroup to the wrongCaller job
+const wrongCallerSgp = new k8s.apiextensions.CustomResource("wrong-caller-sgp", {
+  apiVersion: "vpcresources.k8s.aws/v1beta1",
+  kind: "SecurityGroupPolicy",
+  metadata: {
+    name: "wrong-caller-sgp",
+  },
+  spec: {
+    podSelector: { matchLabels: { app: "wrong-caller" } },
+    securityGroups: {
+      groupIds: [wrongCallerSg.id],
+    }
+  },
+}, { provider: kube });
+
+// Create a job that is not allowed to curl the nginx service. The job will fail if it can reach the service.
+new k8s.batch.v1.Job("wrong-caller", {
+  spec: {
+    template: {
+      metadata: {
+        name: "wrong-caller",
+        labels: {
+          app: "wrong-caller",
+        },
+      },
+      spec: {
+        containers: [{
+          name: "caller",
+          image: "curlimages/curl",
+          command: [
+            "sh", "-c",
+            pulumi.interpolate`curl --silent --show-error --fail ${nginx.metadata.name}.${nginx.metadata.namespace}:80 && exit 1 || exit 0`,
+          ],
+        }],
+        restartPolicy: "Never",
+      },
+    },
+    backoffLimit: 3,
+  },
+}, { provider: kube, dependsOn: [nginx, callerSgp] });
+
+function configureClusterAccess(name: string, cluster: eks.Cluster, sg: aws.ec2.SecurityGroup) {
```

Comment:
Should this live in a helper file?

Reply:
It's an important part of this example (allowing the pods to reach things like coredns). Putting it into another file would hide that aspect IMO

Reply:
Hmm ok. I'm surprised it works while written at the bottom of the file but I guess that's how Node works.